devito/propagator.py

```diff
@@ -191,6 +192,15 @@ def sympy_to_cgen(self, stencils):
         :param stencils: A list of stencils to be converted
         :returns: :class:`cgen.Block` containing the converted kernel
         """
+
+        factors = []
+        if len(self.factorized) > 0:
+            for name, term in zip(self.factorized.keys(), self.factorized):
+                expr = self.factorized[name]
+                # TODO: add support for double precision
+                self.add_local_var(name, "float")
+                # TODO: undo precision enforcing
+                factors.append(cgen.Assign(name, str(ccode(expr.xreplace(self._var_map))).replace("pow", "powf").replace("fabs", "fabsf")))
```

Comment:
The precision enforcing seems like something that should be done to all stencils, no? This would mean we could do it higher up in the constructor, maybe?